### ISSUE TITLE
How to avoid using the OpenAM interface from OpenIDM

### ISSUE THREAD

**Issue by sidkshatriya** [zoltantarcsay/node-openam-agent#18]
Currently the node-openam-agent redirects to OpenAM login screen and once the login is done, the user is redirected back to the demo page.

Very briefly, what might be a best practice if we just want to login to our site without being redirected to OpenAM. There are many REST API calls available and we could probably call them to login to OpenAM -- now how would that integrate with Cookieshield?

Just some hints will be useful. If this question is out of scope for the node-openam-agent -- please go ahead and close the ticket!

**Comment by zoltantarcsay** [comment]
see the noRedirect option for CookieShield at https://github.com/zoltantarcsay/node-openam-agent/wiki/Shield#CookieShield

Alternatively, you can extend CookieShield to use a built in login page.
Another option would be to make the login URL configurable, by overriding https://github.com/zoltantarcsay/node-openam-agent/wiki/PolicyAgent#PolicyAgent+getLoginUrl

**Comment by sidkshatriya** [comment]
Thanks for the hints. This is what I have done and its working:

- Created a custom form for login in my express app at `/login`
- Override the `getLoginUrl` function to simply return `/login` custom login URL at all times
- After user POSTs to the login form, get the session `tokenId` by authenticating via REST api endpoint `openam/json/authenticate`
- Set the cookie `iPlanetDirectoryPro` with the `tokenId` 

Since I set the cookie, cookie shield is able to work seamlessly.

Let me know if there are some other things I need to keep in mind. Thanks again for your help!
